src/libs/lib.c

```diff
@@ -985,11 +985,13 @@ GtkWidget *dt_lib_gui_get_expander(dt_lib_module_t *module)
   gtk_widget_set_halign(hw[DT_MODULE_LABEL], GTK_ALIGN_START);
   gtk_widget_set_halign(hw[DT_MODULE_RESET], GTK_ALIGN_END);
 
-  /* add empty space around widget */
-  gtk_widget_set_margin_start(module->widget, DT_PIXEL_APPLY_DPI(4));
-  gtk_widget_set_margin_end(module->widget, DT_PIXEL_APPLY_DPI(4));
-  gtk_widget_set_margin_top(module->widget, DT_PIXEL_APPLY_DPI(4));
-  gtk_widget_set_margin_bottom(module->widget, DT_PIXEL_APPLY_DPI(4));
+  /* add empty space around module widget
+   * this cannot be set in CSS because the module collapsing is badly handled
+   * */
+  gtk_widget_set_margin_start(module->widget, DT_PIXEL_APPLY_DPI(16));
```

Comment:
Likewise, far too much space lost.

Reply:
Ok, but this defeat the goal to allow all config in CSS and won't make it possible to have a compact theme.

You can remove the hard coded padding (just remove the libe) in the code (imageop.c and lib.c) and add the box in CSS:

```
/* border in module widget, 9pt allows the alignment with module label. */
#iop-plugin-ui box
{
    padding: 5pt 5pt 5pt 9pt;
}

```

Reply:
I tried that, but defining the padding in CSS creates a padding even when the module is collapsed, so you get an empty bubble behind the labels.

Reply:
ok, but now the blending buttons can't be made full-width…

Reply:
Ok, so we need to put the margin in a lower box. Let me check.